src/components/profile/EditProfile.js

```diff
@@ -37,79 +35,70 @@ const EditProfile = ({ screenProps, styles, navigation }) => {
   const [showErrorDialog] = useErrorDialog()
   const { push } = screenProps
 
-  useEffect(() => {
-    if (isEqual(storedProfile, {})) {
-      updateProfile()
-    }
-  }, [])
-
-  // Validate after saving profile state in order to show errors
-  useEffect(() => {
-    //need to pass parameters into memoized debounced method otherwise setX hooks wont work
-    validate()
-  }, [profile])
+  const deboucedProfile = useDebounce(profile, 500)
+  const onProfileSaved = useCallback(() => push(`Dashboard`), [push])
+  const handleEditAvatar = useOnPress(() => push(`ViewAvatar`), [push])
 
   const updateProfile = useCallback(async () => {
     // initialize profile value for first time from storedProfile in userStorage
     const profileFromUserStorage = await userStorage.getProfile()
-    store.set('privateProfile')(profileFromUserStorage)
```

Comment:
this should not be removed, store is global

Reply:
this was kept, just updated to useCurriedSetters hook usage:

```javascript
// line 28
 const [setPrivateProfile] = useCurriedSetters(['privateProfile'])
// ...
// line 46
 setPrivateProfile(profileFromUserStorage)
```